### PR TITLE
mathlib: add Vector3f low pass filter

### PR DESCRIPTION
**Pull request by dagar**
To be used in drivers.

### CODE REVIEW

**Comment by dakejahl** [comment]
Nice. Is there any performance gains by doing this or is it just for usability?

**Comment by dagar** [comment]
TBD, but it's mostly for convenience and not having to duplicate the internal parameters 3 times.

File: src/lib/mathlib/math/filter/LowPassFilter2pVector3f.hpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		// set initial parameters
+		set_cutoff_frequency(sample_freq, cutoff_freq);
+	}
+
+	// Change filter parameters
+	void set_cutoff_frequency(float sample_freq, float cutoff_freq);
+
+	// Add a new raw value to the filter
+	//
+	// @return retrieve the filtered result
```
Comment (bkueng) [inline src/lib/mathlib/math/filter/LowPassFilter2pVector3f.hpp]: - can you change to the typical comments we use `/**`...?
- can you add an explicit `inline` (I know that it's already implicitly requested to be inlined)?

Comment (dagar) [reply src/lib/mathlib/math/filter/LowPassFilter2pVector3f.hpp]: This is the kind of thing where we need a simple style guide or even tool to enforce. I've generally been moving towards c++ style comments, but don't feel strongly about it. I think even doxygen could be used to raise an error.

**Comment by dagar** [comment]
Updated and applied to the MC attitude controller D term LPF for testing.

**Comment by dagar** [comment]
Added a finite check in reset.

**Pull request merged**